tool/net/redbean.c

```diff
@@ -3765,15 +3805,18 @@ static int LuaFetch(lua_State *L) {
       port = usessl ? "443" : "80";
     }
   } else {
-    ip = ntohl(servers.p[0].addr.sin_addr.s_addr);
+    ip = servers.n ? ntohl(servers.p[0].addr.sin_addr.s_addr) : INADDR_LOOPBACK;
     host =
         gc(xasprintf("%hhu.%hhu.%hhu.%hhu", ip >> 24, ip >> 16, ip >> 8, ip));
-    port = gc(xasprintf("%d", ntohs(servers.p[0].addr.sin_port)));
+    port =
```

Comment:
Can the type of port be uint16_t instead of string? You can then use `%d` instead of `%s` below when formatting it into larger values.

Reply:
I considered updating this, but it's used in `getaddinfo`, which requires conversion back to string anyway.